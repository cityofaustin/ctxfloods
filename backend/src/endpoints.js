
var endpoint = 'http://localhost:5000/graphql';

if(process.env.USE_AWS_ENDPOINT) {
  endpoint = process.env.POSTGRAPHQL_ENDPOINT;
};

module.exports = {
    endpoint: endpoint
};
