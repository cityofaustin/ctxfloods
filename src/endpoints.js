
var endpoint = 'http://localhost:5000/graphql';

if(process.env.CI) {
  endpoint = process.env.AWS_ENDPOINT;
};

module.exports = {
    endpoint: endpoint
};
