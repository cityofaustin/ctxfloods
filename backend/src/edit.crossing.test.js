import HttpTransport from 'lokka-transport-http';
import Lokka from 'lokka';
import { endpoint } from './endpoints';

const anonLokka = new Lokka({transport: new HttpTransport(endpoint)});
const superAdminEmail = 'superadmin@flo.ods';
const communityAdminEmail = 'admin@community.floods';
const communityEditorEmail = 'editor@community.floods';
const everyPassword = 'texasfloods';

async function getToken(email, password) {
  const response = await anonLokka.send(`
    mutation($email:String!, $password:String!) {
      authenticate(input: {email: $email, password: $password}) {
        jwtToken
      }
    }
  `,
  {
    email: email,
    password: password
  });

  return response.authenticate.jwtToken;
}

function shouldWork(email, password, communityId, coordinates, extra_description) {
  describe('as ' + email + ' ' + (extra_description || ''), () => {
    var lokka;

    beforeAll(async (done) => {
      getToken(email, password).then((token) => {
        const headers = {
          'Authorization': 'Bearer '+ token
        };
        lokka = new Lokka({transport: new HttpTransport(endpoint, {headers})});
        done();
      });
    });

    var newCrossingId;

    it('should add a new crossing', async () => {
      const response = await lokka.send(`
        mutation($communityId:Int!) {
          newCrossing(input: {
            name: "New Crossing"
            humanAddress: "In test land"
            description: "TEST LAND IS MAGIC!"
            communityId: $communityId
            longitude: -97.755996
            latitude: 30.30718
          }) {
            crossing {
              id
            }
          }
        }
      `,
      {
        communityId: communityId,
      });

      newCrossingId = response.newCrossing.crossing.id;
      expect(response).not.toBeNull();
    });

    it('the new crossing should show up in the DB', async () => {
      const response = await lokka.send(`
        query ($id: Int!) {
          crossingById(id: $id) {
            name
            humanAddress
            description
            communityCrossingsByCrossingId {
              nodes {
                communityId
              }
            }
          }
        }
      `,
      {
        id: newCrossingId
      });

      expect(response).toMatchSnapshot();
    });

    it('should edit the crossing', async () => {
      const response = await lokka.send(`
        mutation editCrossing($id: Int!, $name: String!, $description: String!) {
          editCrossing(input: {crossingId: $id, name: $name, description: $description}) {
            crossing {
              id
              name
              description
            }
          }
        }
      `,
      {
        id: newCrossingId,
        name: "Edited Crossing Name",
        description: "Edited Description"
      });

      expect(response).not.toBeNull();
    });

    it('the edited crossing should show up in the DB', async () => {
      const response = await lokka.send(`
        query ($id: Int!) {
          crossingById(id: $id) {
            name
            humanAddress
            description
            communityCrossingsByCrossingId {
              nodes {
                communityId
              }
            }
          }
        }
      `,
      {
        id: newCrossingId
      });

      expect(response).toMatchSnapshot();
    });

  });
}

describe('When editing a crossing', () => {
  shouldWork(superAdminEmail, everyPassword, 1);
  shouldWork(communityAdminEmail, everyPassword, 1);
  shouldWork(communityEditorEmail, everyPassword, 1);
});
