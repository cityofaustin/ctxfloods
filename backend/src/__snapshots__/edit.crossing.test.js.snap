// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When editing a crossing as admin@community.floods  the edited crossing should show up in the DB 1`] = `
Object {
  "crossingById": Object {
    "communities": Object {
      "nodes": Array [
        Object {
          "id": 1,
        },
      ],
    },
    "description": "Edited Description",
    "humanAddress": "In test land",
    "name": "Edited Crossing Name",
  },
}
`;

exports[`When editing a crossing as admin@community.floods  the new crossing should show up in the DB 1`] = `
Object {
  "crossingById": Object {
    "communities": Object {
      "nodes": Array [
        Object {
          "id": 1,
        },
      ],
    },
    "description": "TEST LAND IS MAGIC!",
    "humanAddress": "In test land",
    "name": "New Crossing",
  },
}
`;

exports[`When editing a crossing as editor@community.floods  the edited crossing should show up in the DB 1`] = `
Object {
  "crossingById": Object {
    "communities": Object {
      "nodes": Array [
        Object {
          "id": 1,
        },
      ],
    },
    "description": "Edited Description",
    "humanAddress": "In test land",
    "name": "Edited Crossing Name",
  },
}
`;

exports[`When editing a crossing as editor@community.floods  the new crossing should show up in the DB 1`] = `
Object {
  "crossingById": Object {
    "communities": Object {
      "nodes": Array [
        Object {
          "id": 1,
        },
      ],
    },
    "description": "TEST LAND IS MAGIC!",
    "humanAddress": "In test land",
    "name": "New Crossing",
  },
}
`;

exports[`When editing a crossing as superadmin@flo.ods  the edited crossing should show up in the DB 1`] = `
Object {
  "crossingById": Object {
    "communities": Object {
      "nodes": Array [
        Object {
          "id": 1,
        },
      ],
    },
    "description": "Edited Description",
    "humanAddress": "In test land",
    "name": "Edited Crossing Name",
  },
}
`;

exports[`When editing a crossing as superadmin@flo.ods  the new crossing should show up in the DB 1`] = `
Object {
  "crossingById": Object {
    "communities": Object {
      "nodes": Array [
        Object {
          "id": 1,
        },
      ],
    },
    "description": "TEST LAND IS MAGIC!",
    "humanAddress": "In test land",
    "name": "New Crossing",
  },
}
`;
